// @sanity/ui/lib/cjs/theme/lib/theme/color/card/createCardStates.js
var __defProp = Object.defineProperty;
var __markAsModule = (target) => __defProp(target, "__esModule", { value: true });
var __export = (target, all) => {
  __markAsModule(target);
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
__export(exports, {
  createCardStates: () => createCardStates
});
function createCardStates(opts, base, dark, name, solid, muted) {
  return {
    enabled: opts.card({
      base,
      dark,
      name,
      state: "enabled",
      solid: solid.default,
      muted: muted.default
    }),
    disabled: opts.card({
      base,
      dark,
      name,
      state: "disabled",
      solid: solid.default,
      muted: muted.default
    }),
    hovered: opts.card({
      base,
      dark,
      name,
      state: "hovered",
      solid: solid.default,
      muted: muted.default
    }),
    pressed: opts.card({
      base,
      dark,
      name,
      state: "pressed",
      solid: solid.default,
      muted: muted.default
    }),
    selected: opts.card({
      base,
      dark,
      name,
      state: "selected",
      solid: solid.default,
      muted: muted.default
    })
  };
}
